pkg/controllers/dynakube/extension/tls.go

```diff
@@ -0,0 +1,112 @@
+package extension
+
+import (
+	"context"
+	"crypto/x509"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/extension/consts"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/certificates"
+	k8slabels "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/labels"
+	k8ssecret "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/secret"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r *reconciler) reconcileTLSSecret(ctx context.Context) error {
```

Comment:
why don't you have a `tls` package, that has a `Reconciler` that can handle the reconcile of this secret following the pattern we already have in the codebase?

Reply:
Good idea, [fixed here](https://github.com/Dynatrace/dynatrace-operator/pull/3778/commits/0b3b87c16ea21c9127cbe963a26c54ad21410e2e)
Unit tests incoming

Reply:
[Unit tests added](https://github.com/Dynatrace/dynatrace-operator/pull/3778/commits/7c824e188f80a032a4823a0e0aabb88002f66b51)